mbf_costmap_nav/src/mbf_costmap_nav/search_helper.cpp

```diff
@@ -0,0 +1,294 @@
+// mbf_costmap_nav
+#include "mbf_costmap_nav/search_helper.h"
+
+// std
+#include <queue>
+#include <unordered_set>
+
+namespace mbf_costmap_nav
+{
+
+SearchHelper::SearchHelper(const costmap_2d::Costmap2DROS* costmap, const SearchConfig& config,
+                           const std::optional<std::function<bool(const Cell, const Cell)>>& compare_strategy,
```

Comment:
Why are the function arguments here passed by value?

Reply:
I don't think there is an improvement in this case because it is only 12 bytes. But it can become bigger in the future, so I will change.